src/components/BooksQuery.tsx

```diff
@@ -1,45 +1,44 @@
-import axios from "axios";
 import { useState } from "react";
 import Card from "@mui/material/Card";
+import { v4 as uuidv4 } from "uuid";
 import "./books.css";
 import { CardContent, Typography } from "@mui/material";
-type book = {
-  id: number;
-  title: string;
-  author: string;
-  year: string;
-};
+import BookObject from "../types/book";
+import AllBooks from "../types/allBooks";
+import ResponseObject from "../types/respBooks";
 const BooksQuery = () => {
   const [book, setBooks] = useState("");
-  const [list, setList] = useState([]);
+  const [list, setList] = useState<AllBooks>({ books: [] });
   const submit = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
     e.preventDefault();
     if (!book) {
       alert("Please enter a book name");
     } else {
-      getData();
+      getData().then((data) => {
+        setList(setBooksModel(data.items));
+      });
     }
-    console.log(list);
   };
-  const getData = () => {
-    axios
-      .get(`https://www.googleapis.com/books/v1/volumes?q=${book}`)
-      .then((data) => {
-        setList(setBooksModel(data.data.items));
-      });
+  const getData = async () => {
+    const response = await fetch(
```

Comment:
use `axios`

and instead of using `axios` directly and repeating the base url multiple times, you should create and export a custom axios instance like this for example

```ts
export const googleBooks = axios.create({
  baseURL: "https://www.googleapis.com/books/v1"
});
```

then you would import this and use it like this

```ts
googleBooks.get<YOUR_RESPONSE_TYPE>("/volumes", {params: ...})
```